unlock-app/src/components/interface/keychain/TransferModal.tsx

```diff
@@ -0,0 +1,226 @@
+import { Button, Input, Modal, Placeholder } from '@unlock-protocol/ui'
+import { useEffect, useState } from 'react'
+import { useForm } from 'react-hook-form'
+import { ToastHelper } from '~/components/helpers/toast.helper'
+import { useAuth } from '~/contexts/AuthenticationContext'
+import { useWeb3Service } from '~/utils/withWeb3Service'
+
+interface TransferModalProps {
+  isOpen: boolean
+  setIsOpen: (state: boolean) => void
+  network: number
+  owner: string
+  tokenId: string
+  expiration: any
+  lock: any
+}
+
+interface TransferFormProps {
+  recipientAddress: string
+  time: number
+}
+
+interface params {
+  keyOwner: string
+  to: string
+  tokenId: string
+  lockAddress: string
+  time?: number
+  lend?: boolean
+}
+
+type TransferOption = 'transfer' | 'share' | 'lend' | undefined
+
+const MAX_TRANSFER_FEE = 10000 //100% of transfer fees = disabled transfers
```

Comment:
I think the logic to check if a key is trsmaferable exists somewhere else. It would be nice to re-use that and (if needed) isolate in some kind of util so we don't re-invent the wheel!

Reply:
> The closest I see to this is isTransferPossible in packages/unlock-js/src/KeyManager.ts, it looks more complex for this specific functionality 🤔

Indeed that is a different thing and indeed I could not find a place where we already have `MAX_TRANSFER_FEE` but please do isliate that logic by creating a `isKeyTransferable` function in a util.